app/decorators/controllers/solidus_gdpr/spree/api/users_controller_decorator.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module SolidusGdpr
+  module Spree
+    module Api
+      module UsersControllerDecorator
+        def emails
+          users_scope = model_class.accessible_by(current_ability, :read)
+          orders_scope = ::Spree::Order.all
+
+          users = users_scope.ransack(params[:q]).result.select(:email)
+          orders = orders_scope.ransack(params[:q]).result(distinct: true).select(:email)
+
+          results = ActiveRecord::Base.connection.execute <<~SQL
+            SELECT emails.* FROM (#{users.to_sql} UNION #{orders.to_sql}) emails
+          SQL
+
+          scope = results.map do |result|
+            {
+              id: result['email'],
+              email: result['email'],
+              orders_count: user_orders_count(result['email'])
+            }
+          end
+
+          @users = Kaminari.paginate_array(scope)
+          respond_with(@users)
+        end
+
+        private
+
+        def user_orders_count(email)
```

Comment:
Unfortunately, this will be a cause of N+1s, because you'd be making a query for each email from the first query. Given the complexity of the first query, it would be a bit complex to get the orders count in a performant way...

I'm wondering whether we should scrap this feature altogether at this point to avoid adding too much complexity. What are your thoughts?